windows/src/desktop/setup/Keyman.Setup.System.ResourceDownloader.pas

```diff
@@ -63,39 +58,36 @@ class function TResourceDownloader.Execute(AInstallInfo: TInstallInfo; ALocation
   end;
 end;
 
-constructor TResourceDownloader.Create(AInstallInfo: TInstallInfo);
+constructor TResourceDownloader.Create(AInstallInfo: TInstallInfo; ASilent: Boolean);
 begin
   inherited Create;
   FInstallInfo := AInstallInfo;
+  FSilent := ASilent;
 end;
 
 function TResourceDownloader.DownloadFile(const ADownloadURL, ADownloadFilename: String): Boolean;
+var
+  frmDownloadProgress: TfrmDownloadProgress;
 begin
   FDownloadURL := ADownloadURL;
   FDownloadFilename := ADownloadFilename;
 
-  { Download the redistributable }
-  frmDownloadProgress := TfrmDownloadProgress.Create(nil);
-  try
-    frmDownloadProgress.Caption := FInstallInfo.Text(ssDownloadingTitle, [ExtractFileName(ADownloadFilename)]);
-    frmDownloadProgress.Callback := DownloadFileCallback;
-    Result := frmDownloadProgress.ShowModal = mrOk;
-  finally
-    frmDownloadProgress.Free;
+  if FSilent then
+    DownloadFileCallback(nil, Result)
+  else
+  begin
+    { Download the redistributable }
+    frmDownloadProgress := TfrmDownloadProgress.Create(nil);
+    try
+      frmDownloadProgress.Caption := FInstallInfo.Text(ssDownloadingTitle, [ExtractFileName(ADownloadFilename)]);
+      frmDownloadProgress.Callback := DownloadFileCallback;
+      Result := frmDownloadProgress.ShowModal = mrOk;
+    finally
+      frmDownloadProgress.Free;
+    end;
   end;
 end;
 
-procedure TResourceDownloader.HTTPCheckCancel(Sender: THttpUploader; var Cancel: Boolean);
-begin
-  frmDownloadProgress.HTTPCheckCancel(Sender, Cancel);
-end;
-
-procedure TResourceDownloader.HTTPStatus(Sender: THttpUploader;
-  const Message: string; Position, Total: Int64);  // I2855
-begin
-  frmDownloadProgress.HTTPStatus(Sender, Message, Position, Total);
-end;
-
```

Comment:
I see these methods were wholesale-removed?  Were they refactored in a previous PR, or is their functionality being dropped?

Reply:
I realised that I could point directly to frmDownloadProgress with only a tiny change to the method signature 😁